vmm/src/vmm_config/machine_config.rs

```diff
@@ -62,6 +71,22 @@ impl Default for VmConfig {
     }
 }
 
+fn validate_vcpu_num<'de, D>(d: D) -> std::result::Result<Option<u8>, D::Error>
```

Comment:
Can you also add a unit test for the path where the vcpu_count is > 32 and check that you are getting the error you expect?

Reply:
done